consensus/reactor.go

```diff
@@ -1,551 +1,561 @@
 	tmevents "github.com/tendermint/tendermint/libs/events"
-	tmjson "github.com/tendermint/tendermint/libs/json"
 	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tendermint/tendermint/libs/service"
 	tmsync "github.com/tendermint/tendermint/libs/sync"
 	"github.com/tendermint/tendermint/p2p"
 	tmcons "github.com/tendermint/tendermint/proto/tendermint/consensus"
 	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
 	sm "github.com/tendermint/tendermint/state"
 	"github.com/tendermint/tendermint/types"
-	tmtime "github.com/tendermint/tendermint/types/time"
+)
+
+var (
+	_ service.Service = (*Reactor)(nil)
+	_ p2p.Wrapper     = (*tmcons.Message)(nil)
+
+	// ChannelShims contains a map of ChannelDescriptorShim objects, where each
+	// object wraps a reference to a legacy p2p ChannelDescriptor and the corresponding
+	// p2p proto.Message the new p2p Channel is responsible for handling.
+	//
+	//
+	// TODO: Remove once p2p refactor is complete.
+	// ref: https://github.com/tendermint/tendermint/issues/5670
+	ChannelShims = map[p2p.ChannelID]*p2p.ChannelDescriptorShim{
+		StateChannel: {
+			MsgType: new(tmcons.Message),
+			Descriptor: &p2p.ChannelDescriptor{
+				ID:                  byte(StateChannel),
+				Priority:            6,
+				SendQueueCapacity:   100,
+				RecvMessageCapacity: maxMsgSize,
+			},
+		},
+		DataChannel: {
+			MsgType: new(tmcons.Message),
+			Descriptor: &p2p.ChannelDescriptor{
+				// TODO: Consider a split between gossiping current block and catchup
+				// stuff. Once we gossip the whole block there is nothing left to send
+				// until next height or round.
+				ID:                  byte(DataChannel),
+				Priority:            10,
+				SendQueueCapacity:   100,
+				RecvBufferCapacity:  50 * 4096,
+				RecvMessageCapacity: maxMsgSize,
+			},
+		},
+		VoteChannel: {
+			MsgType: new(tmcons.Message),
+			Descriptor: &p2p.ChannelDescriptor{
+				ID:                  byte(VoteChannel),
+				Priority:            7,
+				SendQueueCapacity:   100,
+				RecvBufferCapacity:  100 * 100,
+				RecvMessageCapacity: maxMsgSize,
+			},
+		},
+		VoteSetBitsChannel: {
+			MsgType: new(tmcons.Message),
+			Descriptor: &p2p.ChannelDescriptor{
+				ID:                  byte(VoteSetBitsChannel),
+				Priority:            1,
+				SendQueueCapacity:   2,
+				RecvBufferCapacity:  1024,
+				RecvMessageCapacity: maxMsgSize,
+			},
+		},
+	}
 )
 
 const (
-	StateChannel       = byte(0x20)
-	DataChannel        = byte(0x21)
-	VoteChannel        = byte(0x22)
-	VoteSetBitsChannel = byte(0x23)
+	StateChannel       = p2p.ChannelID(0x20)
+	DataChannel        = p2p.ChannelID(0x21)
+	VoteChannel        = p2p.ChannelID(0x22)
+	VoteSetBitsChannel = p2p.ChannelID(0x23)
 
-	maxMsgSize = 1048576 // 1MB; NOTE/TODO: keep in sync with types.PartSet sizes.
+	maxMsgSize = 1048576 // 1MB; NOTE: keep in sync with types.PartSet sizes.
 
 	blocksToContributeToBecomeGoodPeer = 10000
 	votesToContributeToBecomeGoodPeer  = 10000
```

Comment:
What's the role of something like this with respect to the new framework

Reply:
Good question. So in `peerStatsRoutine` we use these variables to invoke the (legacy) `MarkPeerAsGood` API on the switch, which internally does some logic within the address book.

In the new framework, the `PeerManager` has a notion of a "score" for each peer. These scores are currently _binary_ (0 or 100). While it's not clear how exactly to implement a more sophisticated scoring system, I think the idea will be to either still have these APIs or piggy-back off of the `PeerUpdate` mechanism to send scoring updates.

tl;dr `MarkPeerAsGood` -> Update the peer score. How the score is updated remains to be determined.